sdk/storage/storage-file-share/src/Clients.ts

```diff
@@ -994,10 +1039,16 @@ export class ShareClient extends StorageClient {
   ): Promise<ShareGetPropertiesResponse> {
     const { span, spanOptions } = createSpan("ShareClient-getProperties", options.tracingOptions);
     try {
-      return await this.context.getProperties({
+      const res = await this.context.getProperties({
         ...options,
         spanOptions
       });
+
+      // parse protocols
+      const protocols = toShareProtocols(res.enabledProtocols);
+      delete res.enabledProtocols;
+      (res as any).protocols = protocols;
```

Comment:
I don't like `as any`.  What's the benefit of mutating the original result from service? Why not construct and return a new object from it?

Reply:
We can write it in another way to like here. https://github.com/Azure/azure-sdk-for-js/blob/00cb89b703a0100d2fcd370bf2d658f48f267d9e/sdk/storage/storage-blob/src/Clients.ts#L1351

```
   const wrappedRes = {
        ...res,
        _response: res._response, // _response is made non-enumerable
        objectReplicationDestinationPolicyId: res.objectReplicationPolicyId,
        objectReplicationSourceProperties: parseObjectReplicationRecord(res.objectReplicationRules)
      };
```

Since `_response` is made non-enumerable, we have to explicitly copy it. What if there comes another non-enumerable field?
Use `as any` may look bad but is less error-prone.